lib/Sema/MiscDiagnostics.cpp

```diff
@@ -1401,10 +1401,24 @@ static void diagSyntacticUseRestrictions(const Expr *E, const DeclContext *DC,
       Expr *subExpr = nullptr;
       if (calleeName == "??" &&
           (subExpr = isImplicitPromotionToOptional(lhs))) {
+
+        auto lhsStart = reinterpret_cast<const char *>(
+            lhs->getLoc().getOpaquePointerValue());
+        auto dreStart = reinterpret_cast<const char *>(
+            DRE->getLoc().getOpaquePointerValue());
+        auto ptr = dreStart;
+        while (ptr - 1 > lhsStart && std::isspace(*(ptr - 1))) {
```

Comment:
Maybe there is a simpler way to do this, why not take the lhs side end location? It would probably point to before empty spaces?